src/test/java/com/arpnetworking/tsdcore/sinks/DimensionFilteringSinkTest.java

```diff
@@ -86,6 +86,32 @@ public void testInjectExcludeWithout() {
         Mockito.verify(_target).recordAggregateData(dataIncluded);
     }
 
+    @Test
+    public void testWhistlist() {
+        final DimensionFilteringSink sink = new DimensionFilteringSink.Builder()
+                .setName("testWhistlist")
+                .setSink(_target)
+                .setWhitelistDimensions(ImmutableSet.of("foo", "bar"))
+                .build();
+        final PeriodicData dataExcluded = TestBeanFactory.createPeriodicDataBuilder()
+                .setDimensions(ImmutableMap.of("foo", "fooVal", "baz", "bazVal"))
+                .build();
+        sink.recordAggregateData(dataExcluded);
+        Mockito.verify(_target, Mockito.never()).recordAggregateData(Mockito.any(PeriodicData.class));
+
+        final PeriodicData dataIncluded = TestBeanFactory.createPeriodicDataBuilder()
```

Comment:
Missing test case. What happens if the aggregated data has "foo", "bar" and "abc". Based on the current implementation this would get included since it has the white-list dimensions, where it in fact should not because it has an extra dimension. See note above about naming and please include additional testing.